docs/book/basics/simple_resource.md

```diff
@@ -58,8 +58,26 @@ used to configure code generators to run against this code.  The code generators
 generate boilerplate functions and types to complete the Resource definition.
 To learn more on configuring code generation see the *Code Generation* chapter.
 
-Note: The `// +kubebuilder:validation:Pattern=.+:.+` annotation declares Pattern validation
-requiring that the `Image` field match the regular expression `.+:.+`
+**validation**
+Format: `// +kubebuilder:validation:<key=value>`
+The validation annotation supports CRD validation ([OpenAPI v3 schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject)). The last `key-value` part should be a sigle key-value pair. If you have multiple validation key-value pairs, should put them into separte annotation comments.
+Currently, supporting keys are `Maximum`, `Minimum`, `MaxLength`, `MinLength`, `MaxItems`, `MinItems`, `UniqueItems`, `Enum`, `Pattern`, `ExclusiveMaximum`, `ExclusiveMinimum`, `MultipleOf`, `Format`
+The `// +kubebuilder:validation:Pattern=.+:.+` annotation declares Pattern validation requiring that the `Image` field match the regular expression `.+:.+`
+
+**subresource**
+Format:
+`// +kubebuilder:subresource:status`,
+`// +kubebuilder:subresource:scale:specpath=<jsonpath>,statuspath=<jsonpath>,selectorpath=<jsonpath>`
+Subresource annotation support status and scale subresource defined in CRD. The two kinds annotations can be used separately.
+For scale subresource, it contains three fields:`specpath`, `statuspath` and `selectorpath`.
+1) `specpath` refers to `specReplicasPath` attribute of Scale object, and value `jsonpath` defines the JSONPath inside of a custom resource that corresponds to `Scale.Spec.Replicas`. This filed is required and not empty
+2) `statuspath` refers to `statusReplicasPath` attribute of Scale object. and the `jsonpath` value of it defines the JSONPath inside of a custom resource that corresponds to `Scale.Status.Replicas`. This filed is required and not empty
+3) `selectorpath`refers to `labelSelectorPath` attribute of Scale object, and the value `jsonpath` defines the JSONPath inside of a custom resource that corresponds to Scale.Status.Selector. This filed is optional, which can be omitted or with empty value
+
+In order to enable sacle subresource in your modifying type definition file, you have to apply the scale subresource right before the kind struct definition, with correct jsonpath values according to your spec and status. And then make sure the jsonpaths are already defined in the Spec and Status struct. Finally, update the `<kind>_types_test.go` files according to the types Spec and Status changes.
```

Comment:
s/sacle/scale

Reply:
IMO, users look for an example because it helps them visualize the change they need to make
So, consider adding the example snippet.

Reply:
Yes, will add code snippet for illustration.